api/capi/include/nntrainer.h

```diff
@@ -145,6 +145,36 @@ int ml_nnmodel_train_with_generator(ml_nnmodel_h model,
  */
 int ml_nnmodel_destruct(ml_nnmodel_h model);
 
+/**
+ * @brief Get the summary of the neural network model.
+ * @details Use this function to get the summary of the neural network model.
+ * @since_tizen 6.x
+ * @remarks If the function succeeds, @a summary should be released using
+ * free().
+ * @param[in] model The NNTrainer model handler to get summary.
+ * @param[out] summary The summary of the current model. Avoid logic to parse
+ * and exploit @a summary if possible.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_INVALID_PARAMETER Invalid Parameter.
+ */
+int ml_nnmodel_get_summary(ml_nnmodel_h model, char **summary);
```

Comment:
Given that now there is `ml_nnlayer_set_summary`, does this function only provide a summary for only the model or recursively for the layers inside it as well. 
May be add this information to the description.

Reply:
I guess I should add a parameter to control a level of information. Need it be enum or unsigned integer with docs just fine?

Reply:
enum is better, or this can just return the model level summary. For layer summary, the user can just call the layer summary function.

Reply:
How about delete `ml_nnlayer_get_summary` and use this type for a summary?
```cpp
typedef enum {
  ML_TRAIN_SUMMARY_VERBOSE_CONSIZE = 0, /**< Overview of model summary with one-line layer information*/
  ML_TRAIN_SUMMARY_VERBOSE_DETAIL   ,   /**< Detailed model summary with layer properties */
  ML_TRAIN_SUMMARY_VERBOSE_ALL,         /**< Model summary include tensor */
} ml_summary_verbose_type_e;
```

Reply:
thats better. How about rename enum based on what they do - `ML_TRAIN_SUMMARY_MODEL`, `ML_TRAIN_SUMMARY_LAYER`, `ML_TRAIN_SUMMARY_TENSOR`?

Reply:
Yup. I'll add that way